Mathlib/GroupTheory/IndexNormal.lean

```diff
@@ -0,0 +1,80 @@
+/-
+Copyright (c) 2025 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir
+-/
+import Mathlib.Data.Fintype.Perm
+import Mathlib.Data.Nat.Prime.Factorial
+import Mathlib.GroupTheory.Index
+import Mathlib.GroupTheory.Perm.Finite
+
+/-! # Subgroups of small index are normal
+
+* `Subgroup.normal_of_index_eq_smallest_prime_factor`: in a finite group `G`,
+a subgroup of index equal to the smallest prime factor of `Nat.card G` is normal.
+
+* `Subgroup.normal_of_index_two`: in a group `G`, a subgroup of index 2 is normal
+(This does not require `G` to be finite.)
+
+-/
+
+variable {G : Type*} [Group G] {H : Subgroup G} {p : ℕ}
+
+namespace Subgroup
+
+open MulAction MonoidHom Nat
+
+private theorem dvd_prime {r p : ℕ} (hp : p.Prime) (h : r ∣ p !)
+    (hr : ∀ {l : ℕ} (_ : l.Prime) (_ : l ∣ r), p ≤ l) : r ∣ p := by
+  rwa [← Coprime.dvd_mul_right, mul_factorial_pred hp.pos]
+  contrapose! hr
+  obtain ⟨l, hl, hl'⟩ := exists_prime_and_dvd hr
+  rw [dvd_gcd_iff, hl.dvd_factorial, Nat.le_sub_one_iff_lt hp.pos] at hl'
+  exact ⟨l, hl, hl'⟩
+
+/-- A subgroup of index 1 is normal (does not require finiteness of G) -/
+theorem normal_of_index_eq_one (hH : H.index = 1) : H.Normal := by
+  rw [index_eq_one] at hH
+  rw [hH]
+  infer_instance
+
+/-- A subgroup of index 2 is normal (does not require finiteness of G) -/
+theorem normal_of_index_eq_two (hH : H.index = 2) :
+    H.Normal := by
+  obtain ⟨a, ha⟩ := index_eq_two_iff.mp hH
+  have ha1 : ∀ b, b * a ∈ H ↔ ¬ b ∈ H := by simpa only [xor_iff_iff_not] using ha
+  have ha2 : ∀ b, ¬ b * a ∈ H ↔ b ∈ H := by simpa only [xor_iff_not_iff'] using ha
+  refine ⟨fun b hb c ↦ ?_⟩
+  by_cases hc : c ∈ H
+  · exact H.mul_mem (H.mul_mem hc hb) (H.inv_mem hc)
+  rw [← ha1, ← H.inv_mem_iff] at hc
+  rw [← H.inv_mem_iff, ← ha2, ← H.inv_mem_iff, ← ha1] at hb
+  simpa [mul_assoc] using H.mul_mem (H.inv_mem hc) (H.mul_mem hb hc)
```

Comment:
```suggestion
theorem normal_of_index_eq_two (hH : H.index = 2) : H.Normal where
  conj_mem x hxH g := by simp_rw [mul_mem_iff_of_index_two hH, hxH, iff_true, inv_mem_iff]
```

Reply:
Nice! That was the golf I was looking for :)